internal/agent/device/applications/provider/app_handler.go

```diff
@@ -30,19 +33,29 @@ type appTypeHandler interface {
 
 var _ appTypeHandler = (*quadletHandler)(nil)
 var _ appTypeHandler = (*composeHandler)(nil)
+var _ appTypeHandler = (*containerHandler)(nil)
 
 type quadletHandler struct {
 	name           string
 	rw             fileio.ReadWriter
 	volumeProvider volumeProvider
+	specVolumes    *[]v1alpha1.ApplicationVolume
```

Comment:
is this a pointer for any specific reason? a slice is basically a reference type under the hood to an array. also empty slice can be declared at nil.

Reply:
only because it's defined as a pointer on the api object. `handler.specVolumes = provider.Volumes`. I'll change it to not be a pointer and `lo.FromPtr` it